execution/src/posix/execution.cc

```diff
@@ -364,11 +364,12 @@ namespace leatherman { namespace execution {
         }
 
         // Set the locale to C unless specified in the given environment
+        string locale_env;
         if (!environment || environment->count("LC_ALL") == 0) {
-            result.emplace_back("LC_ALL=C");
+            result.emplace_back("LC_ALL="+((inherit && environment::get("LC_ALL", locale_env)) ? locale_env : "C"));
```

Comment:
This isn't quite the behavior we need. If LC_ALL is unset in the parent environment, we should leave it unset in the child process, not set it to C.